jax/_src/linear_util.py

```diff
@@ -347,6 +347,7 @@ def memoized_fun(fun: WrappedFun, *args):
       fun.populate_stores(stores)
     else:
       ans = call(fun, *args)
+      if explain: explain(fun.f, cache is new_cache, cache, key, ans)
```

Comment:
I think we should only call this if the config is switched on. Otherwise this might lead to performance drop.

The logger.DEBUG change is not required if this is only called if the config is on.

Reply:
Another reason is that when we do this for compilation cache by default, it's going to comparing shardings and thousands of devices. For Gemini, we had to carefully optimize things and this would lead to very long lowering times for no reason (since gemini doesn't need this feature).

So it's better to just explain if the config is switched on at WARNING priority.

Reply:
Good thinking!

Reply:
Done!